presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -0,0 +1,164 @@
+import static javax.naming.Context.SECURITY_AUTHENTICATION;
+import static javax.naming.Context.SECURITY_CREDENTIALS;
+import static javax.naming.Context.SECURITY_PRINCIPAL;
+
+public class LdapFilter
+        implements Filter
+{
+    private static final Logger LOG = Logger.get(LdapFilter.class);
+    public static final String AUTHENTICATION_TYPE = "basic";
+    public static final String LDAP_CONTEXT_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+    private final String ldapBaseDn;
+    private final Hashtable<String, String> environment = new Hashtable<>();
+    private DirContext context;
+
+    @Inject
+    public LdapFilter(LdapServerConfig config)
+    {
+        ldapBaseDn = config.getBaseDn();
+
+        environment.put(INITIAL_CONTEXT_FACTORY, LDAP_CONTEXT_FACTORY);
+        environment.put(PROVIDER_URL, config.getLdapUrl());
+        try {
+            context = new InitialDirContext(environment);
+        }
+        catch (NamingException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain nextFilter)
+            throws IOException, ServletException
+    {
+        // skip auth for http
+        if (!servletRequest.isSecure()) {
+            LOG.debug("Skipping LDAP authentication in HTTP.");
+            nextFilter.doFilter(servletRequest, servletResponse);
+            return;
+        }
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+        String header = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        if (header != null) {
+            String[] parts = header.split("\\s+");
+            if (parts.length == 2 && parts[0].equalsIgnoreCase(AUTHENTICATION_TYPE)) {
+                String[] credentials = B64Code.decode(parts[1], StandardCharsets.ISO_8859_1).split(":");
+                if (credentials.length == 2) {
+                    try {
+                        String user = credentials[0];
+                        String password = credentials[1];
+
+                        // TODO: The below code may not work with Active Directory. This is a prototype on openLdap
+                        String dn = getPrincipal(user, context);
+
+                        environment.put(SECURITY_AUTHENTICATION, "simple");
+                        environment.put(SECURITY_PRINCIPAL, dn);
+                        environment.put(SECURITY_CREDENTIALS, password);
+                        context = new InitialDirContext(environment);
+
+                        // ldap authentication ok, continue
+                        nextFilter.doFilter(servletRequest, servletResponse);
+                        return;
+                    }
+                    catch (AuthenticationException e) {
+                        LOG.error("LDAP login failed. Check your userid/password");
```

Comment:
Don't need to log it. It will be logged by the logging filter if you rethrow the exception